Wip 413 kc helm

damienbfs opened the pull request:
- Added chart dependencies on bitnami charts for keycloak and rabbitmq.
- Added setup jobs (keycloak setup and demo setup)
- Added dependencies between jobs/server/worker and keycloak/mongo.

PaulineMauryL commented:
## Coverage report


<img title="Coverage for the whole project is 87.04%. Previous coverage rate is not available, cannot report on evolution." src="https://img.shields.io/badge/Coverage-87%25-orange.svg"> <img title="40% of the statement lines added by this PR are covered" src="https://img.shields.io/badge/PR%20Coverage-40%25-red.svg"><details><summary>Click to see where and how coverage changed</summary><table><thead>
  <tr><th>File</th><th>Statements</th><th>Missing</th><th>Coverage</th><th>Coverage<br>(new stmts)</th><th>Lines missing</th></tr>
</thead>
<tbody><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/administration</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0f">keycloak_admin.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0f"><img title="This PR adds 89 statements to server/lomas_server/administration/keycloak_admin.py. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/89-%28%2B89%29-007ec6.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0f"><img title="This PR adds 73 statements missing coverage to server/lomas_server/administration/keycloak_admin.py. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/73-%28%2B73%29-red.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0f"><img title="The coverage rate of server/lomas_server/administration/keycloak_admin.py is 17.97% (16/89). The file did not seem to exist on the base branch." src="https://img.shields.io/badge/17%25-%2816/89%29-red.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0f"><img title="In this PR, 2 new statements are added to server/lomas_server/administration/keycloak_admin.py, 0 of which are covered (0%)." src="https://img.shields.io/badge/0%25-%280/2%29-red.svg"></a></td><td><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0fR190-R190">190</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-2b1a6a5cf92415e44911bd8daef65ffaab8b7cb9713659a4e57043cbe2236f0fR198-R198">198</a></td><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/administration/scripts</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41d">lomas_demo_setup.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41d"><img title="This PR adds 22 statements to server/lomas_server/administration/scripts/lomas_demo_setup.py. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/22-%28%2B22%29-007ec6.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41d"><img title="This PR adds 3 statements missing coverage to server/lomas_server/administration/scripts/lomas_demo_setup.py. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/3-%28%2B3%29-red.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41d"><img title="The coverage rate of server/lomas_server/administration/scripts/lomas_demo_setup.py is 86.36% (19/22). The file did not seem to exist on the base branch." src="https://img.shields.io/badge/86%25-%2819/22%29-orange.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41d"><img title="In this PR, 3 new statements are added to server/lomas_server/administration/scripts/lomas_demo_setup.py, 2 of which are covered (66.66%)." src="https://img.shields.io/badge/66%25-%282/3%29-red.svg"></a></td><td><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-aebf49d58026285a57d06f4110a497cd9a584a243a4594727844ee414ee0d41dR67-R67">67</a></td></tbody>
<tfoot>
<tr>
<td><b>Project Total</b></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"><img title="This PR adds 4083 statements to the whole project. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/4.1k-%28%2B4083%29-007ec6.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"><img title="This PR adds 529 statements missing coverage to the whole project. The file did not seem to exist on the base branch." src="https://img.shields.io/badge/529-%28%2B529%29-red.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"><img title="The coverage rate of the whole project is 87.04% (3554/4083). The file did not seem to exist on the base branch." src="https://img.shields.io/badge/87%25-%283.6k/4.1k%29-orange.svg"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/255/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"><img title="In this PR, 5 new statements are added to the whole project, 2 of which are covered (40%)." src="https://img.shields.io/badge/40%25-%282/5%29-red.svg"></a></td><td>&nbsp;</td>
</tr>
</tfoot>
</table>

<sub>

This report was generated by [python-coverage-comment-action](https://github.com/py-cov-action/python-coverage-comment-action)

</sub>
</details>



<!-- This comment was produced by python-coverage-comment-action -->

PaulineMauryL commented on .github/workflows/push_docker.yml:
```diff
@@ -4,7 +4,7 @@ on:
     branches: [develop, master]
   push:
-    branches: [develop]
+    branches: [develop, wip_413_kc_helm]
```
remove?

damienbfs replied:
This was just for pushing test images during development. Removed.

PaulineMauryL commented on server/deploy/helm/charts/lomas_server/values.yaml:
```diff
@@ -76,7 +77,7 @@ server:
     annotations:
       nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
-    hostname: "lomas-server.example.com"
+    hostname: "dada-lomas-server.lab.sspcloud.fr"
```
example.com

PaulineMauryL replied:
dada ??

damienbfs replied:
Same, just for developpment. Removed.

PaulineMauryL commented on server/deploy/helm/charts/lomas_server/values.yaml:
```diff
@@ -138,7 +190,7 @@ dashboard:
     annotations:
       nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
-    hostname: "lomas-admin-dashboard.example.ch"
+    hostname: "dada-lomas.lab.sspcloud.fr"
```
modify? below too

PaulineMauryL commented on server/deploy/helm/charts/keycloak/values.yaml:
```diff
@@ -0,0 +1,115 @@
+    # TLS configuration
+    extraTls:
+      - hosts:
+          - "lomas-keycloak-test.lab.sspcloud.fr"
```
also here?

bstuder commented on server/deploy/helm/charts/lomas_server/templates/jobs/lomas_demo_setup_job.yaml:
```diff
@@ -12,6 +12,32 @@ spec:
+                echo "Keycloak not ready yet";
+                sleep 10;
+              done;
+              echo "Keycloak ready, waiting for setup script to complete.."
+              sleep 10;
+              exit 0;
```
do we need the extra 10sec even tho the curl was successful ?

damienbfs replied:
Updated it to specifically wait for the application realm to be available.

bstuder commented on server/deploy/helm/charts/rabbitmq/charts/rabbitmq-15.5.0.tgz:
commit by accident ?

damienbfs replied:
yep, will remove it.

bstuder commented on server/deploy/helm/charts/lomas_server/templates/rabbitmq/configmap.yaml:
so we can delete these ? (and service.yaml / statefulset.yaml) ?

damienbfs replied:
My guess would be yes.

bstuder commented on devenv.nix:
```diff
@@ -637,12 +640,6 @@ in
       exit $pytest_return
     '';
 
-  scripts.run-lomas.exec = ''
-    echo Resetting databases states
-    rm -rf $DEVENV_STATE/{postgres,mongodb}
-    devenv up
-  '';
-
```
nice

damienbfs commented:
A few issues have been created for next steps:
- 427 -> documentation needs to be updated, not only re this pr but also other bits (administration, contributing, etc.)
- 425 -> move sensitive configs to secrets in helm, only once move to pydantic settings done.
- 424 -> implement connection retries for mongoDB from the server/worker so that it does not crash if mongo not yet started.
- 423 -> add loging config to helm

damienbfs committed:
- Added chart dependencies on bitnami charts for keycloak and rabbitmq.
- Added setup jobs (keycloak setup and demo setup)
- Added dependencies between jobs/server/worker and keycloak/mongo.

bstuder commented on server/deploy/helm/charts/lomas_server/templates/jobs/keycloak_setup_job.yaml:
```diff
@@ -0,0 +1,67 @@
+        - name: keycloak-setup
+          image: {{ .Values.admin.image.repository }}:{{ .Values.admin.image.tag }}
+          imagePullPolicy: {{ .Values.admin.image.pullPolicy }}
+          command: ["python", "-m", "lomas_server.administration.scripts.keycloak_setup"]
```
Would it be useful to setup an entry-point in server in [project.scripts] ? like
`keycloak-setup = "lomas_server.administration.scripts.keycloak_setup:kc_setup"
`
Since kc_setup is nicely exposed

damienbfs replied:
Done.

Although we cannot use it in the docker compose, because we do not install the package locally in the dev images (we just set the pythonpath).

bstuder commented on server/deploy/helm/charts/lomas_server/Chart.yaml:
```diff
@@ -13,4 +13,12 @@ icon: "https://raw.githubusercontent.com/dscc-admin-ch/lomas/refs/heads/master/i
     version: "13.18.1"
-    repository: https://charts.bitnami.com/bitnami
+    repository: oci://registry-1.docker.io/bitnamicharts
+  - name: rabbitmq
+    condition: rabbitmq.create
+    version: 15.5.0 # rabbitmq 4.0.8
+    repository: oci://registry-1.docker.io/bitnamicharts
+  - name: keycloak
+    condition: keycloak.create
+    version: 24.4.13 # Keycloak 26.1.4
+    repository: oci://registry-1.docker.io/bitnamicharts
```
nice !

bstuder commented on server/deploy/helm/charts/lomas_server/templates/jobs/lomas_demo_setup_job.yaml:
```diff
@@ -0,0 +1,93 @@
+                sleep 10;
+              done;
+              while true; do
+                STATUS=$(curl -s -o /dev/null -w "%{http_code}" http://{{- .Values.keycloak.fullnameOverride -}}/realms/{{- .Values.admin.realm -}})
```
nice -w trickery ^^

bstuder commented on server/lomas_server/administration/keycloak_admin.py:
```diff
@@ -188,7 +188,8 @@ def del_all_kc_users(kc_config: KeycloakClientConfig) -> None:
     for user in users:
         if (
-            KCAttributeNames.LOMAS_USER_CLIENT in user["attributes"]  # type: ignore
+            "attributes" in user
+            and KCAttributeNames.LOMAS_USER_CLIENT in user["attributes"]  # type: ignore
             and user["attributes"][KCAttributeNames.LOMAS_USER_CLIENT]  # type: ignore
```
no flatmap in python :(, maybe: `user.get('attributes',{}).get(KCAttributeNames.LOMAS_USER_CLIENT)` ?

damienbfs replied:
Yup, makes sense, updated it.

damienbfs committed:
Comments from pr
